### PR TITLE
feat(configurations): configured boilerplate to fastify

### PR DESCRIPTION
**Pull request by melancholiai**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✔                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,31 @@
...
+    libcairo2-dev \
+    libgles2-mesa-dev \
+    libgbm-dev \
+    libllvm7 \
+    libprotobuf-dev \
+    && apt-get -y --purge autoremove \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# COPY . /usr/src/app
```
Comment (syncush) [inline Dockerfile]: remove if not needed

File: Dockerfile [hunk 2]
```diff
@@ -9,24 +35,57 @@ RUN npm install
...
 
-RUN apk add dumb-init
+FROM ubuntu:20.04 AS production
+
+ENV DEBIAN_FRONTEND=noninteractive
 
 ENV NODE_ENV=production
 ENV SERVER_PORT=8080
+ENV CHOKIDAR_USEPOLLING=1
+ENV CHOKIDAR_INTERVAL=500
```
Comment (syncush) [inline Dockerfile]: how does the value affects the service?

File: Dockerfile [hunk 3]
```diff
@@ -9,24 +35,57 @@ RUN npm install
...
+    && apt-get -y --purge autoremove \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# RUN apk add dumb-init
 
 WORKDIR /usr/src/app
 
-COPY --chown=node:node package*.json ./
+# COPY --from=build /usr/src/app /app
```
Comment (syncush) [inline Dockerfile]: remove if not used

File: Dockerfile [hunk 4]
```diff
@@ -9,24 +35,57 @@ RUN npm install
...
 
-COPY --chown=node:node --from=build /tmp/buildApp/dist .
-COPY --chown=node:node ./config ./config
+COPY --from=build /tmp/buildApp/dist .
+COPY ./config ./config
+COPY ./docker-entrypoint.sh .
+COPY ./run.sh .
 
+RUN chmod 777 /usr/src/app/docker-entrypoint.sh
+RUN chmod 777 /usr/src/app/run.sh
```
Comment (syncush) [inline Dockerfile]: merge to one chmod command, chmod accepts multiple file names

`chmod 777 one two three`

File: package.json [hunk 5]
```diff
@@ -70,6 +62,7 @@
     "@types/jest": "^26.0.19",
     "@types/js-yaml": "^3.12.5",
     "@types/multer": "^1.4.5",
+    "@types/node": "^15.0.2",
```
Comment (syncush) [inline package.json]: are you using node `v15.0.2`?  
1. you're `.nvmrc` is declaring node `v12`, we should change it to `v14`
2. you're using the types for node `v15`

File: src/containerConfig.ts [hunk 6]
```diff
@@ -1,26 +1,16 @@
...
-  const tracer = tracing.start();
-  container.register(Services.TRACER, { useValue: tracer });
-
-  const metrics = new Metrics('app_meter');
-  const meter = metrics.start();
-  container.register(Services.METER, { useValue: meter });
   container.register('onSignal', {
     useValue: async (): Promise<void> => {
-      await Promise.all([tracing.stop(), metrics.stop()]);
+      await Promise.all([]);
```
Comment (syncush) [inline src/containerConfig.ts]:
```suggestion
      await Promise.resolve();
```

File: src/index.ts [hunk 7]
```diff
@@ -1,33 +1,19 @@
-/* eslint-disable import/first */
 // this import must be called before the first import of tsyring
 import 'reflect-metadata';
-import { Tracing } from '@map-colonies/telemetry';
-import { createTerminus } from '@godaddy/terminus';
```
Comment (syncush) [inline src/index.ts:5]: why did you remove terminus?

**Review by syncush — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (melancholiai) [reply Dockerfile]: done.

File: Dockerfile (hunk 2 above)
Comment (melancholiai) [reply Dockerfile]: done.

File: Dockerfile (hunk 3 above)
Comment (melancholiai) [reply Dockerfile]: done.

File: Dockerfile (hunk 4 above)
Comment (melancholiai) [reply Dockerfile]: done. there is no need for `run.sh` on the docker anyway

File: package.json (hunk 5 above)
Comment (melancholiai) [reply package.json]: done.

File: src/containerConfig.ts (hunk 6 above)
Comment (melancholiai) [reply src/containerConfig.ts]: done.

File: src/index.ts (hunk 7 above)
Comment (melancholiai) [reply src/index.ts:5]: terminus has no compatibility with fastify, we'll need to use [another package](https://www.fastify.io/ecosystem/) or implement on our own.
at the moment I've implemented a healthcheck plugin, I'm not sure it accomplishes everything we'd like to.

**Review by syncush — APPROVED** [review]

File: src/index.ts (hunk 7 above)
Comment (melancholiai) [reply src/index.ts:5]: my bad. terminus has compatibility with fastify, added terminus instead of a custom plugin.

**Review by syncush — APPROVED** [review]
